Libraries/LibWeb/Internals/Internals.cpp

```diff
@@ -207,4 +207,17 @@ String Internals::get_computed_label(DOM::Element& element)
     return MUST(element.accessible_name(active_document));
 }
 
+void Internals::disable_automatic_page_finish()
+{
+    auto& active_document = internals_window().associated_document();
+    active_document.set_manual_page_load(true);
+}
+
+void Internals::finish_loading()
+{
+    auto& active_document = internals_window().associated_document();
+    active_document.set_manual_page_load(false);
+    active_document.page_did_finish_loading();
+}
```

Comment:
instead of adding new internal functions that override the default test loading behavior, can we change a test runner to always waits for `document.fonts.ready()` to resolve? it’s safe to assume that we always want to load fonts before capturing the ref-test result.

Reply:
@kalenikaliaksandr Thanks for comment! Happy to look into that, but just to be clear. 

What you are suggestion would mean that, I have to modify this:
1. Add a new api in WebContentClient: `did_load_fonts()` and hook this up on both ends.
2. Call `did_load_fonts()` inside [FontFaceSet.cpp::resolve_ready_promise](https://github.com/LadybirdBrowser/ladybird/blob/master/Libraries/LibWeb/CSS/FontFaceSet.cpp#L290) when resolving the ready promise
3. Listen to `on_fonts_loaded()` inside Test.cpp instead of `on_load_finish()`

Or am I missing something/overcomplicating it?

Reply:
@kalenikaliaksandr I tried to do this, but it's really beyond my knowledge on how to get lambda expressions, move/xvalue/temporary values in the compiler right. 
This commit is how far I got: https://github.com/LadybirdBrowser/ladybird/pull/2197/commits/4c1d4aa678fc8a7099f5cab468a88496eea47cc9

The test setup of loading the reference test in the same view as the normal test is a bit funky in relation to how page loads and font loads are being handled.

Is there a way to this in a separate PR and avoid importing the WPT tests in this?